metrics/testing.go

```diff
@@ -22,6 +22,7 @@ const (
 
 // InitForTesting initialize the necessary global variables for unit tests.
 func InitForTesting() {
+	setupMetricsWorker()
```

Comment:
I think this is already called due to `init`?

Reply:
Nope that was really weird because I thought init from the metrics packed (since it is imported) it would be call first but it does not. It fails with a nil pointer exception when it tries to access the mWorker.

Reply:
Cant reproduce it will update.